### PR TITLE
[web][form][SIMS #1093] UI updates for Student Application - Form changes

### PR DESCRIPTION
**Pull request by andrepestana-aot**
- changes to form; 
- changes to confirmModal to allow hiding secondaryButton (missed in the previous PR); 
- added svg image;

image
image
image
image
image
image
image
image

### CODE REVIEW

File: sources/packages/web/src/assets/css/student.scss [hunk 1]
```diff
@@ -36,3 +36,19 @@
...
+  background-image: url("../../assets/images/person-with-clipboard.svg");
+  background-repeat: no-repeat;
+  background-size: contain;
+  width: 100%;
+  height: 100%;
+  min-height: 350px;
+  background-position: center;
+}
+
+.student-start-new-application .v-card {
```
Comment (ann-aot) [inline sources/packages/web/src/assets/css/student.scss:50]: what is this style for?

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue [hunk 2]
```diff
@@ -1,23 +1,31 @@
 <template>
-  <student-page-container>
+  <student-page-container
+    container-class="student-start-new-application"
```
Comment (ann-aot) [inline sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:3]: do we have a property called "container-class" in "student-page-container"
 i

File: sources/packages/web/src/assets/css/formio-shared.scss [hunk 3]
```diff
@@ -69,6 +69,14 @@
   line-height: $line-height-header-lg;
 }
 
+/** 
+*  Inline header to avoid mandatory sign (asterisk) 
+*  to be just after the header and not break the line.
+*/
+.inline-header {
```
Comment (ann-aot) [inline sources/packages/web/src/assets/css/formio-shared.scss:76]: is this responsive?

File: sources/packages/web/src/assets/css/student.scss [hunk 4]
```diff
@@ -36,3 +36,19 @@
...
+  padding-left: 60px !important;
+  padding-right: 60px !important;
+  padding-top: 20 !important;
```
Comment (ann-aot) [inline sources/packages/web/src/assets/css/student.scss:53]: `20px`?

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue [hunk 5]
```diff
@@ -45,6 +54,37 @@
       @pageChanged="pageChanged"
     />
   </student-page-container>
+  <v-row v-if="showNav">
```
Comment (ann-aot) [inline sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:57]: shouldn't  these btns inside "student-page-container"?

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue [hunk 6]
```diff
@@ -45,6 +54,37 @@
...
+    </v-col>
+    <v-col>
+      <v-btn
+        class="float-right"
+        color="primary"
+        v-show="!isLastPage"
+        @click="wizardGoNext"
+        >Next step</v-btn
+      >
+      <v-btn
```
Comment (ann-aot) [inline sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:76]: mm.. bit confused. we already have a submit btn at line 25. then why this btn?

File: sources/packages/web/src/assets/css/student.scss (hunk 1 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/assets/css/student.scss:50]: Sorry, forgot to remove that.

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue (hunk 2 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:3]: No, it doesn't. I forgot to remove that. Thanks!

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue [hunk 7]
```diff
@@ -1,23 +1,31 @@
...
+            v-if="!notDraft && !isFirstPage && !submittingApplication"
             variant="outlined"
             :loading="savingDraft"
             @click="saveDraft()"
           >
             {{ savingDraft ? "Saving..." : "Save draft" }}</v-btn
           >
           <v-btn
-            v-if="!isReadOnly"
+            v-if="!isReadOnly && !isFirstPage"
```
Comment (ann-aot) [inline sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:26]: also as part of refactor, we can remove `<v-progress-circular` at line 32 and use `loading` property of the btn

File: sources/packages/web/src/assets/css/formio-shared.scss (hunk 3 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/assets/css/formio-shared.scss:76]: It is just to keep the asterisk  connected to label. Otherwise the asterisk will go to the next line. It won't be any different for small screens.

File: sources/packages/web/src/assets/css/student.scss (hunk 4 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/assets/css/student.scss]: removed. Thanks!

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue (hunk 5 above)
Comment (ann-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:54]: are these btns working? I see same btns in `sources/packages/web/src/components/common/StudentApplication.vue`

**Review by ann-aot — COMMENTED** [review]
will you be adding other program year forms in this PR or a different PR?

**Review by ann-aot — COMMENTED** [review]
mm, i didn't understand this 
image
why is this disabled?

**Comment by ann-aot** [comment]
I know its not part of the PR, but just wondering why this was added `<!DOCTYPE html>` @andrewsignori-aot  @dheepak-aot  @guru-aot . any idea?
image

**Comment by ann-aot** [comment]
we no longer use well, 
image.
we use panels with `formio-panel-bg-secondary` customclass

**Comment by ann-aot** [comment]
do we need `"attr": "alt",`?
image

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue (hunk 5 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:54]: Yeah, actually we don't need those buttons here. Thanks!

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue (hunk 6 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:73]: Removed those buttons, thanks!

**Comment by andrepestana-aot** [comment]
> mm, i didn't understand this image why is this disabled?

Yeah, I don't think we need that Custom Property. I removed that. Thanks!

**Comment by andrepestana-aot** [comment]
> formio-panel-bg-secondary

ok. Thanks! Changed to panel.

**Comment by andrepestana-aot** [comment]
> do we need `"attr": "alt",`? image

No, we don't. I've added to check something and forgot to remove it. Thanks!

**Comment by andrewsignori-aot** [comment]
> I know its not part of the PR, but just wondering why this was added `<!DOCTYPE html>` @andrewsignori-aot @dheepak-aot @guru-aot . any idea? image

I have no idea. @ann-aot, it is hard to follow the comments in this way. Please find any point in the code to start the discussion from there.

File: sources/packages/web/src/components/common/StudentApplication.vue [hunk 8]
```diff
@@ -18,12 +18,15 @@
         @click="wizardGoPrevious"
         >Previous section</v-btn
       >
+      <v-btn color="primary" v-show="isFirstPage" @click="wizardGoNext"
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/common/StudentApplication.vue:21]: The navigation buttons can be probably replaced by the `<footer-button>` component with the `justify="space-between"`.
The navigation button on the right using the `@click="wizardGoNext"` can be probably the same with the label defined by the current page.

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue [hunk 9]
```diff
@@ -1,39 +1,38 @@
...
+      </header-navigator>
+    </template>
     <body-header>
       <template #actions>
         <v-row class="m-0 p-0 float-right">
           <v-btn
             color="primary"
-            v-if="!notDraft"
-            v-show="!isFirstPage && !submittingApplication"
+            v-if="!notDraft && !isFirstPage && !submittingApplication"
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:15]: Why was it changed?

Comment (andrepestana-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:15]: Both v-if and v-show are there to show/hide the button. For me it is hard to understand if you have 2 places to check the logic. Also I prefer using v-if when possible cause it doesn't render the component in case of false. v-show renders it and hides it in case of false, which makes the component to flash in the screen when rendering.

**Comment by andrepestana-aot** [comment]
> will you be adding other program year forms in this PR or a different PR?

When everything is done for the 2023-2024 version I can just copy and change what is different.

**Comment by ann-aot** [comment]
> > I know its not part of the PR, but just wondering why this was added `<!DOCTYPE html>` @andrewsignori-aot @dheepak-aot @guru-aot . any idea? image
> 
> I have no idea. @ann-aot, it is hard to follow the comments in this way. Please find any point in the code to start the discussion from there.

the file is too big to open in GitHub and comment on. that's why I opened it in `https://github.dev/` and pasted the screenshots. is there any other way to open and comment big files

File: sources/packages/web/src/assets/css/student.scss [hunk 10]
```diff
@@ -36,3 +36,13 @@
   height: 100%;
   min-height: 350px;
 }
+
+.student-with-clipboard-img {
+  background-image: url("../../assets/images/person-with-clipboard.svg");
```
Comment (dheepak-aot) [inline sources/packages/web/src/assets/css/student.scss:41]: @extend .img-background?

File: sources/packages/web/src/components/common/StudentApplication.vue (hunk 8 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/components/common/StudentApplication.vue]: Thanks, I used footer-buttons.

File: sources/packages/web/src/assets/css/student.scss (hunk 10 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/assets/css/student.scss]: yeah, ok. Moved it to formio-shared.scss close to the others.

File: sources/packages/web/src/components/common/modals/ConfirmModal.vue [hunk 11]
```diff
@@ -16,6 +16,7 @@
         @primaryClick="resolvePromise(true)"
         @secondaryClick="resolvePromise(false)"
         :disablePrimaryButton="disablePrimaryButton"
+        :showSecondaryButton="showSecondaryButton"
```
Comment (ann-aot) [inline sources/packages/web/src/components/common/modals/ConfirmModal.vue:19]: what is the scenario, where we will not show the "cancel/back" Secondary Button? for my info

File: sources/packages/web/src/components/common/StudentApplication.vue [hunk 12]
```diff
@@ -269,6 +271,7 @@ export default {
     };
 
     const submitted = (args: any, form: any) => {
+      processing.value = true;
```
Comment (ann-aot) [inline sources/packages/web/src/components/common/StudentApplication.vue:274]: when this property is set to false? what happens to the loading when there is an error in submission?

File: sources/packages/web/src/components/common/modals/ConfirmModal.vue (hunk 11 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/components/common/modals/ConfirmModal.vue:19]: It's for this scenario:
image

File: sources/packages/web/src/components/common/StudentApplication.vue (hunk 12 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/components/common/StudentApplication.vue]: Yes, you're right. I created a `processing` prop and get it from the parent component. Thanks!

File: sources/packages/web/src/assets/css/formio-shared.scss (hunk 3 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/assets/css/formio-shared.scss:76]: Just to clear this out:
image

This class will keep the asterisk glued to the label but the label will wrap the lines as needed.

Comment (ann-aot) [reply sources/packages/web/src/assets/css/formio-shared.scss:76]: okay. Thank you for the clarification. that was my comment

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue (hunk 9 above)
Comment (guru-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:15]: I feel we have either one of them to avoid confusion.

File: sources/packages/web/src/components/common/StudentApplication.vue [hunk 13]
```diff
@@ -8,27 +8,28 @@
...
+  <footer-buttons
+    justify="space-between"
+    :processing="processing"
+    @secondaryClick="wizardGoPrevious"
+    :showSecondaryButton="!isFirstPage"
+    secondaryLabel="Previous section"
+    class="mx-0"
+  >
+    <template #primary-buttons="{ disabled }">
+      <span>
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/common/StudentApplication.vue:20]: Is there a need to use the template for the primary button for this case?
I believe that the out-of-box properties are enough to do the personalization needed. If I am missing something please let me know.

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue (hunk 9 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:15]: `v-if`` and `v-show` serves different purposes. I would be ok to not use them but since they were in place it would be a downgrade.
There is a technical reason why both were used but I am not sure if there is a technical reason to remove them.
 Since it is a minor issue I will not ask to change it back.

Comment (andrewsignori-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:15]: > I feel we have either one of them to avoid confusion.

The confusion can be avoided with a quick check of the documentation.

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue [hunk 14]
```diff
@@ -1,39 +1,38 @@
...
-              indeterminate
-              color="secondary"
-              size="23"
-            />
             {{ submittingApplication ? "Submitting..." : "Submit application" }}
           </v-btn>
         </v-row>
       </template>
     </body-header>
+
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:35]: Any particular reason why the blank lines were added here and in line 48?

File: sources/packages/web/src/views/student/financial-aid-application/StudentApplication.vue [hunk 15]
```diff
@@ -10,39 +10,41 @@
...
-          class="ma-2"
-          variant="elevated"
-          data-cy="primaryFooterButton"
-          color="primary"
-          @click="startApplication"
-          >Start Application</v-btn
-        >
-      </content-group>
-    </v-form>
+    <v-card class="mt-4 p-4 w-100">
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/student/financial-aid-application/StudentApplication.vue:13]: The `v-card` is also present in the `student-page-container`, usually, there is no need for one to be added.

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue (hunk 9 above)
Comment (ann-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:15]: `v-if and v-show` have different purposes. But I don't remember the reason here.

**Comment by andrewsignori-aot** [comment]
> > > I know its not part of the PR, but just wondering why this was added `<!DOCTYPE html>` @andrewsignori-aot @dheepak-aot @guru-aot . any idea? image
> > 
> > 
> > I have no idea. @ann-aot, it is hard to follow the comments in this way. Please find any point in the code to start the discussion from there.
> 
> the file is too big to open in GitHub and comment on. that's why I opened it in `https://github.dev/` and pasted the screenshots. is there any other way to open and comment big files

@ann-aot I got the part where we were not able to add the comments there, IMO we can get any other point in the next file and add the comments there.

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue (hunk 9 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:15]: Keep on one way or another and let's move on 😉 
There will be no impact and we can shorten the discussion.

File: sources/packages/web/src/components/common/StudentApplication.vue (hunk 13 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/components/common/StudentApplication.vue:20]: Ok. Removed the template. Thanks!

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue (hunk 14 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue]: Removed. Thanks!

File: sources/packages/web/src/views/student/financial-aid-application/StudentApplication.vue (hunk 15 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/views/student/financial-aid-application/StudentApplication.vue]: ok. Used LayoutTemplates.CenteredCard. Thanks!

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes, looks good 👍

**Comment by ann-aot** [comment]
@andrepestana-aot i see some disabled fields in the form, what is the purpose of that, 
image
image
image

**Comment by andrepestana-aot** [comment]
> @andrepestana-aot i see some disabled fields in the form, what is the purpose of that, image image image

They are going to be displayed when the form is in read only mode. Instead of the select, it renders their labels as text. It was already this way when I started working on it.

I see the institution one had a wrong label and position and fixed that. Thanks!
